core/lib/via_btc_client/src/indexer/mod.rs

```
@@ -1 +1,252 @@
+            }
+            Message::L1BatchDAReference(m) => {
+                if let Ok(sender) = Self::get_sender_address(&m.common) {
+                    sender == self.sequencer_address
+                } else {
+                    false
+                }
+            }
+            Message::ProofDAReference(m) => {
+                if let Ok(sender) = Self::get_sender_address(&m.common) {
+                    sender == self.sequencer_address
+                } else {
+                    false
+                }
+            }
+            Message::L1ToL2Message(m) => m.amount > bitcoin::Amount::ZERO,
+            Message::SystemBootstrapping(_) => true,
+        }
+    }
+
+    fn process_bootstrap_message(
+        state: &mut BootstrapState,
+        message: Message,
+    ) -> BitcoinIndexerResult<()> {
+        match message {
+            Message::SystemBootstrapping(sb) => {
+                state.verifier_addresses = sb.input.verifier_addresses;
+                state.bridge_address = Some(sb.input.bridge_p2wpkh_mpc_address);
+                state.starting_block_number = sb.input.start_block_height;
+            }
+            Message::ProposeSequencer(ps) => {
+                if let Ok(sender) = Self::get_sender_address(&ps.common) {
+                    if state.verifier_addresses.contains(&sender) {
+                        state.proposed_sequencer = Some(ps.input.sequencer_p2wpkh_address);
+                    }
+                }
+            }
+            Message::ValidatorAttestation(va) => {
+                if state.proposed_sequencer.is_some() {
+                    if let Ok(sender) = Self::get_sender_address(&va.common) {
+                        if state.verifier_addresses.contains(&sender) {
+                            state.sequencer_votes.insert(sender, va.input.attestation);
+                        }
+                    }
+                }
+            }
+            _ => {
+                return Err(BitcoinError::Other(
```

Should we return an error here or just ignore the non-bootstrap related messages?

---

fair point